src/test/java/com/google/sps/ChatServletTest.java

```diff
@@ -0,0 +1,258 @@
+        String expected = tofu.newRenderer(CHAT_TEMPLATE).setData(templateData).render();
+        String actual = stringWriter.getBuffer().toString().trim();
+
+        Assert.assertEquals(expected, actual);
+    }
+
+    // @Test
+    // public void servletPostGoodComment() throws IOException {
+    //     // POST method should post the message to datastore and
+    //     // redirect to /chat
+
+    //     when(request.getParameter(MESSAGE_TEXT_PROPERTY)).thenReturn("hi");
+    //     when(request.getParameter(GROUP_ID_PARAM)).thenReturn(GROUP_ID);
+    //     when(response.getWriter()).thenReturn(printWriter);
+
+    //     servlet.doPost(request, response);
+
+    //     // Only one message ("hi") should be in the datastore
+    //     String expected = "hi";
+    //     Query messageQuery = new Query(MESSAGE_KIND + GROUP_ID);
+    //     Entity onlyMessageEntity = datastore.prepare(messageQuery).asSingleEntity();
+    //     String actual = (String) onlyMessageEntity.getProperty(MESSAGE_TEXT_PROPERTY);
+
+    //     verify(response, times(1)).sendRedirect("/chat");
+    //     Assert.assertEquals(expected, actual);
+    // }
+
+    // @Test
+    // public void servletPostToxicComment() throws IOException {
+    //     // POST method should not post to datastore and
+    //     // should render an error message
+
+    //     when(request.getParameter(MESSAGE_TEXT_PROPERTY)).thenReturn(MESSAGE_TEXT_TOXIC);
+    //      when(request.getParameter(GROUP_ID_PARAM)).thenReturn(GROUP_ID);
+    //     when(response.getWriter()).thenReturn(printWriter);
+
+    //     servlet.doPost(request, response);
+
+    //     String expectedMessages = null;
+    //     Query messageQuery = new Query(MESSAGE_KIND + GROUP_ID);
+    //     // Should be null
+    //     Entity actualMessages = datastore.prepare(messageQuery).asSingleEntity();
+
+    //     ImmutableMap<String, String> errorData = ImmutableMap.of("errorMessage", 
+    //     "Your message contains content that may be deemed offensive by others. " +
+    //     "Please revise your message and try again.");
+
+    //     String expectedOutput = tofu.newRenderer("templates.chat.error").setData(errorData).render();
+    //     String actualOutput = stringWriter.getBuffer().toString().trim();
+        
+    //     Assert.assertEquals(expectedMessages, actualMessages);
+    //     Assert.assertEquals(expectedOutput, actualOutput);
+    // }
+
+    @Test
+    public void servletPostEmptyComment() throws IOException {
+        // POST method should not post to datastore but should
+        // redirect to /chat
+
+        when(request.getParameter(MESSAGE_TEXT_PROPERTY)).thenReturn("");
+        when(request.getParameter(GROUP_ID_PARAM)).thenReturn(GROUP_ID);
+        when(response.getWriter()).thenReturn(printWriter);
+
+        servlet.doPost(request, response);
+
+        String expectedMessages = null;
+        Query messageQuery = new Query(MESSAGE_KIND + GROUP_ID);
+        // Should be null
```

Comment:
To make this clearer: 
  - Remove this comment
  - Change the assert to be `Assert.assertEquals(null, actualMessages);`